Source/utils/language.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <string>
+
+#include <fmt/format.h>
+
+#include "utils/log.hpp"
+
+namespace devilution {
+
+// Local definition to fix compilation issue due to header conflict.
+// TODO: remove this once #1735 has been merged.
+[[noreturn]] void app_fatal(const char *pszFmt, ...);
+
+namespace language {
+
+void Initialize();
+std::string Translate(const std::string &key);
+
+} // namespace language
+
+template <typename... Args>
+std::string _(const std::string &key, Args &&... args)

```

Comment:
This signature means a new string is created for every `_` call. We probably don't want it like this, the `_` should only return the translated but not yet interpolated format string (like it did before).

Reply:
Since there are arguments that will be passed to `fmt::format` there will be a new `std::string` created in any case. The inline version of `_` that does not take arguments should not though (but I'll need to double-check that this version is actually called).

This form is only to simplify having translations with arguments, replacing
```
fmt::format(_("You have {} gold"), gold)
```
with
```
_("You have {} gold", gold)
```

But both forms will always allocate a new `std::string` because of the format anyway so I don't see the issue.

Reply:
Many strings do not accept an argument but having a different return type for the no-arg version would be somewhat confusing.

If a string does not accept an argument, you end up copying it needlessly.

Moreover, even if a string does accept an argument, there is no longer a way to store just the key without creating a new string.

Reply:
Fair enough, I'll remove that format integration from `_` then.